### PR TITLE
Misc2

### PR DESCRIPTION
**Pull request by arose**
- Performance tweaks
    - Make SymmetryOperator.createMapping monomorphic
    - Improve bonding-sphere calculation
    - Defer Scene properties calculation (markerAverage, opacityAverage, hasOpaque)
    - Improve checks in in UnitsRepresentation setVisualState
- Add RepresentationRegistry.clear and ThemeRegistry.clear

### CODE REVIEW

File: src/mol-gl/scene.ts [hunk 1]
```diff
@@ -311,12 +315,24 @@ namespace Scene {
                 return boundingSphereVisible;
             },
             get markerAverage() {
+                if (markerAverageDirty) {
+                    markerAverage = calculateMarkerAverage();
+                    markerAverageDirty = true;
```
Comment (dsehnal) [inline src/mol-gl/scene.ts:320]:
```suggestion
                    markerAverageDirty = false;
```

File: src/mol-gl/scene.ts [hunk 2]
```diff
@@ -311,12 +315,24 @@ namespace Scene {
...
+                }
                 return markerAverage;
             },
             get opacityAverage() {
+                if (opacityAverageDirty) {
+                    opacityAverage = calculateOpacityAverage();
+                    opacityAverageDirty = true;
```
Comment (dsehnal) [inline src/mol-gl/scene.ts:327]:
```suggestion
                    opacityAverageDirty = false;
```

File: src/mol-gl/scene.ts [hunk 3]
```diff
@@ -311,12 +315,24 @@ namespace Scene {
...
+                }
                 return opacityAverage;
             },
             get hasOpaque() {
+                if (hasOpaqueDirty) {
+                    hasOpaque = calculateHasOpaque();
+                    hasOpaqueDirty = true;
```
Comment (dsehnal) [inline src/mol-gl/scene.ts:334]:
```suggestion
                    hasOpaqueDirty = false;
```

File: src/mol-gl/scene.ts (hunk 1 above)
Comment (dsehnal) [reply src/mol-gl/scene.ts:320]: Unless I am missing something.

File: src/mol-math/geometry/symmetry-operator.ts [hunk 4]
```diff
@@ -177,11 +177,15 @@ namespace SymmetryOperator {
 
     export interface Coordinates { x: ArrayLike<number>, y: ArrayLike<number>, z: ArrayLike<number> }
 
-    export function createMapping<T extends number>(operator: SymmetryOperator, coords: Coordinates, radius?: ((index: T) => number)): ArrayMapping<T> {
-        const invariantPosition = SymmetryOperator.createCoordinateMapper(SymmetryOperator.Default, coords);
-        const position = operator.isIdentity ? invariantPosition : SymmetryOperator.createCoordinateMapper(operator, coords);
+    export function createMapping<T extends number>(operator: SymmetryOperator, coords: Coordinates, radius: ((index: T) => number)): ArrayMapping<T> {
```
Comment (dsehnal) [inline src/mol-math/geometry/symmetry-operator.ts:180]: While small, this is a breaking change and has ugly API. How about:

```ts
function _createMapping(...) { ... }

export function createMapping<T extends number>(operator: SymmetryOperator, coords: Coordinates, radius: ((index: T) => number) = _zeroRadius) {
  return _createMapping(...);
}
```

Comment (arose) [reply src/mol-math/geometry/symmetry-operator.ts]: sounds good, done

**Pull request merged**